pkg/endpoint/policy.go

```diff
@@ -286,6 +286,9 @@ func (e *Endpoint) SetIdentity(owner Owner, id *policy.Identity) {
 	defer tree.Mutex.Unlock()
 	cache := owner.GetConsumableCache()
 
+	e.SecLabel = id
+	e.Consumable = cache.GetOrCreate(id.ID, id)
+
```

Comment:
This can't be right. The code is the way it is to remove an eventual existing consumable on an endpoint from the cache and then add the new one.